src/sentry/integrations/discord/client.py

```diff
@@ -133,44 +146,65 @@ def track_response_data(
-                discord_error_code = str(discord_error_response.get("code", ""))
-                if discord_error_code in DISCORD_ERROR_CODES:
-                    code_message = DISCORD_ERROR_CODES[discord_error_code]
-                # These are excluded since they are not actionable from our side
-                if discord_error_code in DISCORD_USER_ERRORS:
-                    include_in_slo = False
-            except Exception:
-                pass
+        if not is_ok or error:
+            self._handle_failure(log_params=log_params, resp=resp)
+        else:
+            self._handle_success(log_params=log_params)
+
+    def _handle_failure(
+        self,
+        log_params: dict[str, Any],
+        resp: Response | None = None,
+    ) -> None:
+        """
+        Do extra logic to handle an error from Discord
+        """
 
+        discord_error_response: dict | None = None
+        if resp is not None:
+            # Try to get the additional error code that Discord sent us to help determine what specific error happened
+            try:
+                discord_error_response = json.loads(resp.content.decode("utf-8"))
+                log_params["discord_error_response"] = discord_error_response
+            except Exception as err:
+                self.logger.info(
+                    "error trying to handle discord error message", exc_info=err, extra=log_params
+                )
+
+        discord_error_code = None
+        if discord_error_response is not None:
+            # Discord sends us a special code for errors in the response data
+            # https://discord.com/developers/docs/topics/opcodes-and-status-codes#json
+            discord_error_code = str(discord_error_response.get("code", ""))
+            log_params["discord_error_code"] = discord_error_code
+
+            # Get the specific meaning for those codes
+            if discord_error_code_message := DISCORD_ERROR_CODES.get(discord_error_code, None):
+                log_params["discord_error_code_message"] = discord_error_code_message
+
+        # Check if the error is due to a user configuration error, which we do not have control over to fix
+        # An example of this would be if the user deleted the discord guild and never updated the alert action
+        is_user_error = discord_error_code in DISCORD_USER_ERRORS
+        log_params["is_user_error"] = is_user_error
+
+        metrics_key = self._METRICS_USER_ERROR_KEY if is_user_error else self._METRICS_FAILURE_KEY
         metrics.incr(
-            f"{self.metrics_prefix}.http_response",
+            metrics_key,
             sample_rate=1.0,
-            tags={
-                str(self.integration_type): self.name,
-                "status": code,
-                "is_ok": is_ok,
-                "include_in_slo": include_in_slo,
-                "discord_code": discord_error_code,
-            },
+            tags=log_params,
```

Comment:
won't this create a lot of cardinality, and are we concerned about costs? we are potentially adding tags for a lot of different things (`is_user_error`, `discord_error_code_message`, `discord_error_code`, `discord_error_response`, `logging_context`, `str(self.integration_type)`)

Reply:
oh if we were already emitting this i guess it's fine

Reply:
It's a good call out though, do we pay per tag?

Reply:
removed the tags as they don't provide any additional context at this moment